lib/Mojo/IOLoop/ReadWriteFork.pm

```diff
@@ -359,6 +387,18 @@ sub _read {
   return unless defined $read;
   return unless $read;
   warn "[$self->{pid}] >>> @{[ESC($buffer)]}\n" if DEBUG;
+  if (my $timer = $self->_read_timeout_id){
```

Comment:
that code maybe should be modified to first check whether read_timeout is set and only then to the other stuff
 
```perl
if (my $timeout = $self->read_timeout){
    if (my $timer = $self->_read_timeout_id){
        $self->ioloop->reactor->again($timer);
    }
    else {
        Scalar::Util::weaken($self);
        $self->_read_timeout_id(
            $self->ioloop->timer($timeout => sub {
                $self->emit('read_timeout');
                $self->_read_timeout_id(undef);
            })
        );
    }
}
```